liblangutil/DebugInfoSelection.h

```diff
@@ -0,0 +1,86 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Handles selections of debug info components.
+ */
+
+#pragma once
+
+#include <map>
+#include <optional>
+#include <ostream>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace solidity::langutil
+{
+
+/**
+ * Represents a set of flags corresponding to components of debug info selected for some purpose.
+ *
+ * Provides extra functionality for enumerating the components and serializing/deserializing the
+ * selection to/from a comma-separated string.
+ */
+struct DebugInfoSelection
```

Comment:
This might look a bit over-engineered on its own but I'm planning to make it a template and reuse it for other, similar options. It's based on `SolverChoice`, which already has most of this functionality. We have a lot of CLI options that represent a bunch of boolean flags parsed from comma-separated lists:
- [`SolverChoice`](https://github.com/ethereum/solidity/blob/v0.8.7/libsmtutil/SolverInterface.h#L39-L107)
- [`ModelCheckerEngine`](https://github.com/ethereum/solidity/blob/v0.8.7/libsolidity/formal/ModelCheckerSettings.h#L115-L142)
- [`ModelCheckerTargets`](https://github.com/ethereum/solidity/blob/v0.8.7/libsolidity/formal/ModelCheckerSettings.h#L92-L113)
- [`CombinedJsonRequests`](https://github.com/ethereum/solidity/blob/v0.8.7/solc/CommandLineParser.h#L76-L98)
- [`CompilerOutputs`](https://github.com/ethereum/solidity/blob/v0.8.7/solc/CommandLineParser.h#L54-L74)

Mine is the sixth and we might want to add even more in the future.

A big problem with all of them is that when adding a new flag, forgetting to include it in one place will not cause an error, just give you wrong results. My version solves it by defining a mapping between names and members and reusing it in all definitions that need to handle all members.